app-emulation/virtualbox-guest-additions/virtualbox_guest_additions-6.1.26.recipe

```diff
@@ -0,0 +1,109 @@
+SUMMARY="VirtualBox kernel modules and user-space tools for Haiku guests"
+DESCRIPTION="VirtualBox is a free x86 virtualization solution allowing a wide \
+range of x86 operating systems such as Windows, DOS, BSD or Linux to run on a \
+Linux system.
+
+This package contains the VirtualBox kernel modules and user-space \
+tools for Haiku guests."
+HOMEPAGE="https://www.virtualbox.org/"
+COPYRIGHT="2003-2017 Oracle Corporation"
+LICENSE="
+	GNU GPL v2
+	CDDL v1
+	"
+REVISION="1"
+SOURCE_URI="http://download.virtualbox.org/virtualbox/$portVersion/VirtualBox-$portVersion.tar.bz2"
+CHECKSUM_SHA256="0212602eea878d6c9fd7f4a3e0182da3e4505f31d25f5539fb8f7b1fbe366195"
+SOURCE_DIR="VirtualBox-$portVersion"
+PATCHES="virtualbox_guest_additions-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
```

Comment:
One last request for now :)
Could you change the following? This is after a recent change in Haikuporter: https://github.com/haikuports/haikuporter/commit/da92250cd2ac73492e9b094d1eb561ecca9fd771
```diff
- ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
+ ARCHITECTURES="all !x86_gcc2"
```

Reply:
I looked at that change and it confused me.  Does all imply it works on all architectures but the ones specified with !xxx?

It's not really true at all for this package, if anything it should just be ARCHITECTURES="x86_64"?

Reply:
@waddlesplash maybe you could elaborate a bit here?

Reply:
If it really only works on x86_64, then yes, that is the only thing that should be mentioned.

Reply:
OK, then maybe `!x86_gcc2` isn't needed in `ARCHITECTURES`?

Reply:
My understanding is Virtualbox isn't supported outside of x86 architectures.  Currently it won't be any use on any 32-bit Haiku system either.  So only x86_64 is supported at all.  

If I'm understanding the change made to haikuports correctly, then all is a way to say everything is supported, and then we can explicitly list things that aren't supported as we encounter them or if we know already they are not.

So going forward more generally, it sounds like we're no longer doing the ?arch thing, marking all on everything and then marking everything we know won't be supported with !arch.

Reply:
Recipes that are highly architecture-specific can avoid using `all` if it makes sense to and use the "old way", and this one probably qualifies.

Reply:
Essentially, in addition to the "opt-in" model, there is now an "opt-out" model, and the "opt-out" model is now the default.

Reply:
Thanks @waddlesplash